### DISCUSSION TITLE
Vertico and completion-preview mode

### DISCUSSION THREAD

**Discussion by Doodler8888** [minad/vertico#512] [Q&A] [answered]
There is a new addition in emacs 30 called 'completion-preview mode', which allows to see completion suggestions as some sort of an echo. And it renders incorrectly using vertico completion:

![image](https://github.com/user-attachments/assets/155e2024-6ed9-419c-bb76-e7959f845b18)

There is no such problem using a minibuffer with a non-interactive completion nor ivy:

![image](https://github.com/user-attachments/assets/a6f9abae-77ef-42f1-befc-362272e0c5c6)

 To reproduce, evaluate '(add-hook 'minibuffer-mode-hook #'completion-preview-mode)'

GNU Emacs 30.0.60 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.43, cairo version 1.18.0) of 2024-08-20

**Comment by minad** [comment, marked as answer]
Hi! Using these two modes together in the minibuffer is not recommended, and as it seems, is also not supported right now. There is almost no advantage of enabling `completion-preview-mode` inside the minibuffer. It only adds overhead and leads to interfering key bindings. Also the second example you are showing is misleading. While Vertico prioritizes `config.org`, `completion-preview-mode` suggests `colors` as completion. My recommendation is to only enable `completion-preview-mode` in non-`completing-read` minibuffers.

**Reply by Doodler8888** [reply]
Yeah, i enabled it this way for the eval, shell-command and evil-mode cmd line minibuffers. And i didn't know it's quite easy to enable the mode more selectively when it comes to them.

```
(defun my-conditional-completion-preview ()
  "Enable completion-preview-mode selectively."
  (if (or (eq this-command 'eval-expression)
          (eq this-command 'shell-command)
          (eq this-command 'evil-ex))
      (completion-preview-mode 1)
    (completion-preview-mode -1)))

(add-hook 'minibuffer-setup-hook #'my-conditional-completion-preview)
```
